Source/Search/Filters.lua

```diff
@@ -0,0 +1,154 @@
+--
+--  Auctionator.Search.Filters is an empty table on load, need to populate
+--  with the possible filters
+--
+--  Here's what one entry looks like:
+--  {
+--    classID    = integer (corresponding to ITEM_CLASS_IDS )
+--    name       = string  (resolved by GetItemClassInfo( classID ))
+--    filter     = table   (new QueryAuctionItems filterData format, { classID, subClassID (nil), inventoryType (nil) } )
+--    subClasses = {
+--      classID  = integer (subClassID)
+--      name     = string  (resolved by GetItemSubClassInfo( subClassID ))
+--      filter   = table   (new QueryAuctionItems filterData format, { classID, subClassID, inventoryType (nil) } )
+--      TODO: Probably want to use the inventoryType to create Armor slot filters as well...
+--    }
+--  }
+
+local ITEM_CLASS_IDS = {
+  LE_ITEM_CLASS_WEAPON,
+  LE_ITEM_CLASS_ARMOR,
+  LE_ITEM_CLASS_CONTAINER,
+  LE_ITEM_CLASS_GEM,
+  LE_ITEM_CLASS_ITEM_ENHANCEMENT,
+  LE_ITEM_CLASS_CONSUMABLE,
+  LE_ITEM_CLASS_GLYPH,
+  LE_ITEM_CLASS_TRADEGOODS,
+  LE_ITEM_CLASS_RECIPE,
+  LE_ITEM_CLASS_BATTLEPET,
+  LE_ITEM_CLASS_QUESTITEM,
+  LE_ITEM_CLASS_MISCELLANEOUS
+}
+local INVENTORY_TYPE_IDS = {
+  LE_INVENTORY_TYPE_HEAD_TYPE,
+  LE_INVENTORY_TYPE_SHOULDER_TYPE,
+  LE_INVENTORY_TYPE_CHEST_TYPE,
+  LE_INVENTORY_TYPE_WAIST_TYPE,
+  LE_INVENTORY_TYPE_LEGS_TYPE,
+  LE_INVENTORY_TYPE_FEET_TYPE,
+  LE_INVENTORY_TYPE_WRIST_TYPE,
+  LE_INVENTORY_TYPE_HAND_TYPE,
+}
+
+Auctionator.Search.Filter = {
+  classID = 0,
+  name = Auctionator.Constants.FilterDefault,
+  key = 0,
+  parentKey = nil,
+  filter = {},
+  subClasses = {}
+}
+
+-- XXX: Ununsed in current code (was used in advanced search dialog)
```

Comment:
Make `XXX` a `TODO` (my IDE flags these for review for me 😄 ). Also, typo in unused.

Reply:
My editor flags XXX and TODO 😉